base_layer/core/src/transactions/transaction_protocol/transaction_initializer.rs

```diff
@@ -272,8 +272,9 @@ impl SenderTransactionInitializer {
             1 => RecipientInfo::Single(None),
             _ => RecipientInfo::Multiple(HashMap::new()),
         };
-        let mut ids = Vec::with_capacity(self.num_recipients);
-        for i in 0..self.num_recipients {
+        let num_ids = max(1, self.num_recipients);
```

Comment:
I don't agree with this. Doesn't it potentially break some assertions down the line? I feel like the fact that this builder was called with num_recipients = 0 is the problem. If a Tx has zero recipients then how can there be a TxID?

Reply:
This might be true, but other methods such as the create_tx from the base layer core make use of zero recipients. From this I believe, the number of recipients specifies the number of external recipients and when specifying 0 recipients it will construct a transaction that spends to self. When more than one recipient is specified then extra outputs are added to the fee calculation when it determines if a change output should be added when building using the SenderTransactionInitializer. Also, when the recipients are set to zero then the transaction can immediately be finalized, otherwise it enters the SenderState::SingleRoundMessageReady state that needs additional steps to finalize. If I remember correctly it requires a full serialization and deserialization of the partial transaction using the current mechanisms to get it finalized.

Reply:
Ok, that does make sense. I guess an update to the docstring or argument names to specify that recipients refer to an external party might make it clearer then what it means to have 0 recipients because on face value a MW tx with zero recipients doesn't sound right.